src/coreComponents/codingUtilities/StringUtilities.cpp

```diff
@@ -63,6 +63,44 @@ RETURN_TYPE tokenize( string const & str,
 
   return tokens;
 }
+/**
+ * String tokenizing by whitespace function
+ **/
+template< typename RETURN_TYPE >
+RETURN_TYPE tokenizeBySpaces( string const & str )
+{
+  if( str.empty() )
+  {
+    return {};
+  } else {
+    const size_t strEnd=str.size();
+    RETURN_TYPE tokens;
+    size_t pos=0;
+
+    while (pos<strEnd) {
+      //search for a token begining
+      while ( pos<strEnd && ( std::isspace( str[pos] ) ) )
+      {
+        ++pos;
+      }
+
+      if (pos<strEnd)
+      {
+        size_t lastPos=pos;
+
+        //search for the token ending
+        do
+        {
+          ++pos;
+        } while ( pos<strEnd && !std::isspace( str[pos] ) );
+
+        tokens.emplace_back( str.substr( lastPos, pos - lastPos ) );
+      }
+    }
+    
+    return tokens;
+  }
+}
```

Comment:
Would
```c++
std::vector< string > tokenizeBySpaces( string const & str )
{
    return tokenize( str, " \f\n\r\t\v", true );
}
```
be equivalent? I know you're also dealing with the last value being void, but you can surely manage this as a post processing of the `tokenize` call with something like `if ( std::isspace( str.back() ) and not result.empty() ) result.pop_back();`?

Reply:
In Python the idiomatic way to split a string with leading/trailing whitespace would be something like `s.strip().split()`, with the `split()` part already handled by our `tokenize()`, so maybe we could also just implement `strip()`?

Reply:
Yes, you're most probably correct, trimming the input is surely a better way.

Reply:
Currently, `tokenize("1|2|3", "|", true)` returns 3 results and `tokenize("|1|2|3|", "|", true)` returns 4 results with the first one being empty (where we could intuitively expect 3 or 5 results with empty entries at the sides).

I'm wondering, would it be be better to :
- Assume this behaviour is expected (and maybe document this on `tokenize()`) and just pre-trim the input of `tokenizeBySpaces()`, as @klevzoff suggested ?
- Adapt the `tokenize()` algorithm so it never create empty entries when treatConsecutiveDelimAsOne is on (and maybe rename this parameter to "dontCreateEmptyEntries", so its behaviour will be clearer) ?

Reply:
> Currently, `tokenize("1|2|3", "|", true)` returns 3 results and `tokenize("|1|2|3|", "|", true)` returns 4 results with the first one being empty (where we could intuitively expect 3 or 5 results with empty entries at the sides).

Yep, that's clearly a bug if so. I would expect 5 results here, for parity with `str.split()` functions in Python, JS and numerous other languages. Sometimes empty substrings *are* significant, so I don't think we would want 3 results by default. Although we could add another boolean parameter to control this?

Reply:
As these options behavior would be exclusive, would an enum with three values be acceptable ? (treatEveryDelims, treatConsecutiveDelimAsOne, dontCreateEmptyEntries)

Reply:
> Currently, tokenize("1|2|3", "|", true) returns 3 results and tokenize("|1|2|3|", "|", true) returns 4 results with the first one being empty (where we could intuitively expect 3 or 5 results with empty entries at the sides).

We surely do not want this 4 entry result... 3 seems legit, while 5 can be acceptable. But 4 makes little sense.
I was thinking about the `Group::getGroupByPath` member function (which tokenizes on `/` with a precise usage of an initial `/` to define an absolute path), but it does not use the `tokenize` function, so it's OK.

I would personally go for the solutions that returns 5 tokens (that way, there's a difference between `1|2|3` and `|1|2|3|`), with maybe the possibility to trim (as an option?) the input string and remove the potentially first and last dummy values.

Reply:
> As these options behavior would be exclusive, would an enum with three values be acceptable ? (treatEveryDelims, treatConsecutiveDelimAsOne, dontCreateEmptyEntries)

Are they? The input string `|a||b|` could be converted into
- `a`, `b`
- `a`, `-`, `b`
- `-`, `a`, `b`, `-`
- `-`, `a`, `-`, `b`, `-`

which are 4 choices?

Reply:
To sum up, the solution will be to adapt tokenize to have `tokenize( string const & str, string const & delimiters, bool const treatConsecutiveDelimAsOne=true, bool const preTrimStr=false )`.

Following this logic, `||a||b||` would give : 
- tokenize( [...], 1, 1 ) = `a`,`b`
- tokenize( [...], 0, 1 ) = `a`,`-`,`b`
- tokenize( [...], 1, 0 ) = `-`,`a`,`b`,`-`
- tokenize( [...], 1, 1 ) = `-`,`-`,`a`,`-`,`b`,`-`,`-`